packages/git/src/node/init/git-init.ts

```diff
@@ -53,27 +53,31 @@ export class DefaultGitInit implements GitInit {
 
     async init(): Promise<void> {
         const { env } = process;
-        if (typeof env.LOCAL_GIT_DIRECTORY !== 'undefined' || typeof env.GIT_EXEC_PATH !== 'undefined') {
-            await this.handleExternalNotFound('Cannot use Git from the PATH when the LOCAL_GIT_DIRECTORY or the GIT_EXEC_PATH environment variables are set.');
-        } else {
-            try {
-                const { execPath, path, version } = await findGit();
-                if (!!execPath && !!path && !!version) {
-                    // https://github.com/desktop/dugite/issues/111#issuecomment-323222834
-                    // Instead of the executable path, we need the root directory of Git.
-                    const dir = dirname(dirname(path));
-                    const [execPathOk, pathOk, dirOk] = await Promise.all([pathExists(execPath), pathExists(path), pathExists(dir)]);
-                    if (execPathOk && pathOk && dirOk) {
-                        process.env.LOCAL_GIT_DIRECTORY = dir;
-                        process.env.GIT_EXEC_PATH = execPath;
-                        this.logger.info(`Using Git [${version}] from the PATH. (${path})`);
-                        return;
+        try {
+            const { execPath, path, version } = await findGit();
+            if (!!execPath && !!path && !!version) {
+                // https://github.com/desktop/dugite/issues/111#issuecomment-323222834
+                // Instead of the executable path, we need the root directory of Git.
+                const dir = dirname(dirname(path));
+                const [execPathOk, pathOk, dirOk] = await Promise.all([pathExists(execPath), pathExists(path), pathExists(dir)]);
+                if (execPathOk && pathOk && dirOk) {
+                    if (typeof env.LOCAL_GIT_DIRECTORY !== 'undefined' && env.LOCAL_GIT_DIRECTORY !== dir) {
+                        this.logger.error(`Misconfigured env.LOCAL_GIT_DIRECTORY: ${env.LOCAL_GIT_DIRECTORY}. dir was: ${dir}`);
+                        await this.handleExternalNotFound('The LOCAL_GIT_DIRECTORY env variable was already set to a different value.');
                     }
+                    if (typeof env.GIT_EXEC_PATH !== 'undefined' && env.GIT_EXEC_PATH !== execPath) {
+                        this.logger.error(`Misconfigured env.GIT_EXEC_PATH: ${env.GIT_EXEC_PATH}. execPath was: ${execPath}`);
```

Comment:
isn't it more of a warning that the configured `GIT_EXEC_PATH` will be ignored as `git` was found on the `PATH`?

Reply:
> Fails when the env variable was set, and we try to set another value.

Maybe I'm just confused. Should it fail or just yell?

Reply:
> `GIT_EXEC_PATH` will be ignored as `git` was found on the `PATH`?

`GIT_EXEC_PATH` is a different thing:
```
which git && `which git` --exec-path
/usr/local/bin/git
/usr/local/Cellar/git/2.20.1/libexec/git-core
```

Reply:
> yell